toonz/sources/toonz/xsheetdragtool.cpp

```diff
@@ -1761,12 +1761,18 @@ class VolumeDragTool final : public XsheetGUI::DragTool {
   void onDrag(const QMouseEvent *event) override {
     if (!m_enabled) return;
 
-    const Orientation *o   = getViewer()->orientation();
-    QRect track            = o->rect(PredefinedRect::VOLUME_TRACK);
-    NumberRange range      = o->frameSide(track);
-    int frameAxis          = o->frameAxis(event->pos());
-    double v               = range.ratio(frameAxis);
-    if (o->flipVolume()) v = 1 - v;
+    const Orientation *o = getViewer()->orientation();
+    QRect track          = o->rect(PredefinedRect::VOLUME_TRACK);
+    NumberRange range    = o->frameSide(track);
+    int frameAxis        = o->frameAxis(event->pos());
+    if (o->isVerticalTimeline() &&
+        !o->flag(PredefinedFlag::VOLUME_AREA_VERTICAL)) {
+      range     = o->layerSide(track);
+      frameAxis = o->layerAxis(event->pos());
```

Comment:
For now, with "Classic-revised" or "Compact" layout, you cannot control the volume if you put the sound level in other than the first column.
Should this line be like as follows ? :
```
-      frameAxis = o->layerAxis(event->pos());
+      frameAxis = o->layerAxis(event->pos()) - getViewer()->columnToLayerAxis(m_index);
```

Reply:
Good catch.  Fixed.